gocourse18/dbreader.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+func connectToDatabase(dsn string) (*gorm.DB, error) {
+	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to database: %w", err)
+	}
+	return db, nil
+}
+
+func getUsers(db *gorm.DB) ([]User, error) {
+	var users []User
+	if err := db.Find(&users).Error; err != nil {
```

Comment:
Робота з БД - це чудовий приклад для застосування контексту:
```suggestion
	if err := db.WithContext(ctx).Find(&users).Error; err != nil {
```

https://go.dev/doc/database/cancel-operations
https://gorm.io/docs/context.html

Reply:
Попрошу додати контекст у всі функції, які читають і які записують в БД.